test/knowledge_panels_from_json_test.dart

```diff
@@ -422,4 +427,67 @@ void main() {
     KnowledgePanels kp = KnowledgePanels.fromJson(panels);
     expect(kp.panelIdToPanelMap.length, equals(8));
   });
+
+  // Ensure that a "text_element" with some sources data is well parsed
+  test('Load knowledge panel JSON with a text element', () {
+    final String id = 'additive_en:e262';
+
+    final Map<String, dynamic> jsonData = <String, dynamic>{
+      id: {
+        'elements': [
+          {
+            'element_type': 'text',
+            'text_element': {
+              'html':
+              '<strong>Sodium acetate:</strong> Sodium acetate, CH3COONa, also abbreviated NaOAc, is the sodium salt of acetic acid. This colorless deliquescent salt has a wide range of uses.',
+              'source_language': 'English',
+              'source_lc': 'en',
+              'source_text': 'Wikipedia',
+              'source_url': 'https://en.wikipedia.org/wiki/Sodium acetate'
+            }
+          }
+        ],
+        'level': 'info',
+        'size': 'small',
+        'title_element': {'title': 'E262 - Sodium acetates'},
+        'topics': ['health']
+      },
+    };
+
+    KnowledgePanels kp = KnowledgePanels.fromJson(jsonData);
+    expect(kp.panelIdToPanelMap.length, equals(1));
+
+    expect(
+      kp.panelIdToPanelMap[id]!.elements!.length,
+      equals(1),
+    );
+    expect(
+      kp.panelIdToPanelMap[id]!.elements!.first.elementType,
```

Comment:
I'm sure you could put `kp.panelIdToPanelMap[id]!.elements!.first` in a variable instead of repeating it 6 times.